lib/asset_writer.rb

```diff
@@ -0,0 +1,37 @@
+class AssetWriter
+  attr_reader :asset, :target_path
+
+  delegate :logical_path, :digest_path, :to => :asset
+
+  def initialize(asset_or_path, target_dir)
+    @asset = find_asset(asset_or_path)
+    @target_dir = target_dir
+    @target_path = Rails.root.join(target_dir, @asset.digest_path)
```

Comment:
Does this need to be so closely tied to the application?  I feel like this should be closer to lib/pdf_generator.  Should the caller care what the path is?

I'm thinking that the whole PDF Generator could become a gem, and would require this to go with it.  Do you have other use cases in mind for this class?